state-chain/pallets/cf-validator/src/lib.rs

```diff
@@ -801,6 +874,216 @@ pub mod pallet {
+				Error::<T>::AlreadyAssociatedWithOperator
+			);
+			ensure!(
+				T::AccountRoleRegistry::has_account_role(&validator_id, AccountRole::Validator),
+				Error::<T>::NotValidator
+			);
+			ClaimedValidators::<T>::append(&validator_id, &operator_id);
+			Self::deposit_event(Event::ValidatorClaimed {
+				validator: validator_id,
+				operator: operator_id,
+			});
+			Ok(())
+		}
+
+		/// Executed by a validator to accept an operator's invitation to manage it.
+		#[pallet::call_index(11)]
+		#[pallet::weight(T::ValidatorWeightInfo::accept_operator())]
+		pub fn accept_operator(origin: OriginFor<T>, operator_id: T::AccountId) -> DispatchResult {
+			let validator_id = T::AccountRoleRegistry::ensure_validator(origin)?;
+			ensure!(
+				ManagedValidators::<T>::get(&validator_id).is_none(),
+				Error::<T>::AlreadyAssociatedWithOperator
+			);
+
+			let claimed_by_operators =
+				ClaimedValidators::<T>::take(&validator_id).ok_or(Error::<T>::NotClaimed)?;
+
+			ensure!(claimed_by_operators.contains(&operator_id), Error::<T>::OperatorDoesNotMatch);
+
+			ManagedValidators::<T>::insert(&validator_id, &operator_id);
+
+			Self::deposit_event(Event::AcceptedOperatorByValidator {
+				validator: validator_id,
+				operator: operator_id,
+			});
+
+			Ok(())
+		}
+
+		/// Executed by an operator or an validator to remove itself from the operator's delegation
+		/// pool.
+		#[pallet::call_index(12)]
+		#[pallet::weight(T::ValidatorWeightInfo::remove_validator())]
+		pub fn remove_validator(origin: OriginFor<T>, validator: T::AccountId) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+			let operator =
+				ManagedValidators::<T>::get(&validator).ok_or(Error::<T>::ValidatorDoesNotExist)?;
+			ensure!(account_id == operator || account_id == validator, Error::<T>::NotAuthorized);
+			ManagedValidators::<T>::remove(&validator);
+
+			Self::deposit_event(Event::ValidatorHasBeenRemoved { validator, operator });
+
+			Ok(())
+		}
+
+		/// Executed by an operator to set the delegation preferences for the operator.
+		#[pallet::call_index(13)]
+		#[pallet::weight(T::ValidatorWeightInfo::set_delegation_preferences())]
+		pub fn set_delegation_preferences(
+			origin: OriginFor<T>,
+			preferences: DelegationPreferences,
+		) -> DispatchResult {
+			let operator_id = T::AccountRoleRegistry::ensure_operator(origin)?;
+			OperatorParameters::<T>::insert(&operator_id, preferences.clone());
+			Self::deposit_event(Event::DelegationPreferencesSet {
+				operator: operator_id,
+				preferences,
+			});
+			Ok(())
+		}
+
+		/// Executed by an operator to block a delegator from delegating to the operator.
+		#[pallet::call_index(14)]
+		#[pallet::weight(T::ValidatorWeightInfo::block_delegator())]
+		pub fn block_delegator(origin: OriginFor<T>, delegator_id: T::AccountId) -> DispatchResult {
+			let operator_id = T::AccountRoleRegistry::ensure_operator(origin)?;
+
+			// TODO: Check of delegator is currently delegation to this operator and error if so.
```

Comment:
I don't think it would be an error - we would just remove the delegator->operator mapping.